### PR TITLE
try to fix master/slave ip

### LINKED ISSUE #11: Add port forwardings for Spark and HDFS web apps, e.g. 8080, 4040, ...
_From @hannes-ucsc on April 18, 2015 7:24_

_Copied from original issue: BD2KGenomics/cgcloud-spark#2_

### PR DESCRIPTION
**Pull request by hyviquel**
partially fix #11

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by hannes-ucsc** [comment]
This is great, we were just talking about this issue in our team.

@cket, you might want to incorporate the SPARK_PUBLIC_DNS fix into the Toil appliance.

File: spark/src/cgcloud/spark/spark_box.py [hunk 1]
```diff
@@ -272,9 +272,11 @@ def __install_spark( self ):
             SPARK_LOCAL_DIRS=self._lazy_mkdir( spark_dir, "local" ),
             JAVA_HOME='/usr/lib/jvm/java-7-oracle',
             SPARK_MASTER_IP='spark-master',
-            HADOOP_CONF_DIR=fmt( "{install_dir}/hadoop/etc/hadoop" ) )
+            HADOOP_CONF_DIR=fmt( "{install_dir}/hadoop/etc/hadoop" ),
+            SPARK_PUBLIC_DNS="$public_name" )
         with remote_open( spark_env_sh_path, use_sudo=True ) as spark_env_sh:
             spark_env_sh.write( '\n' )
+            spark_env_sh.write( 'public_name=$(curl -s http://169.254.169.254/latest/meta-data/public-hostname)\n' )
```
Comment (hannes-ucsc) [inline spark/src/cgcloud/spark/spark_box.py:279]: Is the indirection via public_name necessary? If `$public_name` works, `$(curl …)` should, too, right?

**Comment by hannes-ucsc** [comment]
@cket, never mind. I confused Spark and Mesos. 8-o

**Comment by hannes-ucsc** [comment]
Test failure looks like a fluke.

Jenkins, test this please.

File: spark/src/cgcloud/spark/spark_box.py (hunk 1 above)
Comment (hyviquel) [reply spark/src/cgcloud/spark/spark_box.py:279]: mmh I'm not sure, I will try to test when find time ;-)

**Comment by hannes-ucsc** [comment]
Continued in #232

**Pull request closed without merging**